galley/pkg/config/analysis/analyzers/injection/injection-version.go

```diff
@@ -96,43 +110,24 @@ func (a *VersionAnalyzer) Analyze(c analysis.Context) {
 			if v == "" {
 				continue
 			}
-			// Note the pod/version to check later after we've collected all injector versions.
-			podVersions = append(podVersions, podVersion{
-				Resource:     r,
-				ProxyVersion: v})
 
+			if v != injectionVersion {
```

Comment:
I was going to add a comment saying "Don't forget to handle the case where the "sidecar.istio.io/proxyImage" doesn't match the running pod, but now that I think about it, that shouldn't happen - if the annotation is applied then the injector runs immediately (as it's on the resource that's being mutated by the webhook).

Does that sound right to you?

Reply:
Yes, the image is updated to the image specified in the annotation. Plus we are skipping pods with that annotation in this analyzer.